### DISCUSSION TITLE
How to solve complex PDE (i.e with imaginary part) with MOOSE?

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#26711] [Q&A General] [answered]
Hi, I was wondering if this is possible and was curious how to go about it. Say I have a PDE of the form

iw*u - d^2/dx^2(u) = 0 (but in 3D)

so an expression with both complex and real parts. How could I go about solving it in MOOSE? The FEM discretization will still have a complex number attached. Would I run the real and complex parts independently? 

Since I'm coupling this with an optimization problem (i.e I am trying to optimize material properties) that could be an issue. I guess I would need to optimize outside of MOOSE with a script and then re-do the simulations each time. Or what other options do I have?

**Comment by GiudGiud** [comment]
Hello

There's been a few posts on that lately. I think the main idea is to create two equations that will end up being coupled
- one for the real part of the equation
- one for the imaginary part

We have an optimization module in moose that is integrated and can work with several equations. I dont think you ll need to use an external optimizer necessarily

Guillaume

**Reply by richmondodufisan** [reply]
I guess this is more of a math question than a MOOSE one but let's say the expression was:

iw*u - d^2/dx^2(u) = f(u)

Would the two PDEs be:

iw*u = Im(f(u) .......................(1)

- d^2/dx^2(u) = Re(f(u)) ...................(2)

with the overall solution being

u = Re(u) + Im(u) 

Did I get that right? But even still, u in both expressions can be complex valued.

**Reply by GiudGiud** [reply, marked as answer]
no I think you need to replace u with X (Re(u)  and i Y. (Im(u))

so 
w X -d2x Y = Im(f)
and
-w Y - d2x X = Re(f)

**Reply by richmondodufisan** [reply]
Ahh I see! That makes sense, thank you.

**Comment by lynnmunday** [comment]
We do something like this here:  https://mooseframework.inl.gov/modules/tensor_mechanics/1d_elastic_waves.html
